### PR TITLE
docs: templates update

### PR DESCRIPTION
**Pull request by zmzlois**
## What's added in this PR

- configuring custom templates

### CODE REVIEW

**Review by ivan-burazin — CHANGES_REQUESTED** [review]
@zmzlois styles ( src/styles/_typography.scss) cannot be in the same PR as content, please create 2 PRs

File: src/content/docs/configuration/templates.mdx [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+import Keyboard from "../../../components/Keyboard.astro"
+import { Tabs, TabItem } from "@astrojs/starlight/components"
+
+Templates are displayed on the right side of your Daytona dashboard when you try to create a [workspace](/usage/workspaces). You can use them to customize your Daytona dashboard in combination with [devcontainer](https://www.daytona.io/dotfiles/advanced-configuration-techniques-for-dev-container) during deployment to create workspaces faster based on your requirements.
+
+We will walk through how to customize them, create workspaces and repositories based on templates, to deploy your own templates and the default templates on the Daytona dashboard.
+
+<hr/>
+
+## Prerequisites
```
Comment (ivan-burazin) [inline src/content/docs/configuration/templates.mdx:17]: @zmzlois this is unnecessary. especially sin you state in the paragraph below _"we assume that you either have a successful Daytona deployment and gained access to a Daytona dashboard URL, or you have Daytona CLI installed on your local machine."_

File: src/content/docs/configuration/templates.mdx [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+You can use Daytona CLI to create workspaces and repositories based on templates. If you haven't installed it yet, you can run the following commands in your terminal:
+
+```bash
+brew install daytonaio/daytona/daytona
+```
+Read more about [Daytona CLI and how to use it](/tools/cli).
+
+<hr/>
+
+## Creating workspaces from templates
```
Comment (ivan-burazin) [inline src/content/docs/configuration/templates.mdx:34]: @zmzlois this should not be part of CONFIGURATION at all. You can if you wish note about creating from template and forward them to this section: https://www.daytona.io/docs/usage/workspaces/

File: src/content/docs/configuration/templates.mdx [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+You can use the arrow keys to select one.
+:::
+</TabItem>
+</Tabs>
+
+<hr/>
+
+##### Optional: Connecting your workspace to a new Git repository
+
+To facilitate CI/CD, you might need to reset repository URL upon template creation.
```
Comment (ivan-burazin) [inline src/content/docs/configuration/templates.mdx:107]: This is about usage and has no place in creation of templates

File: src/content/docs/configuration/templates.mdx [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+git merge --allow-unrelated-histories origin/main
+
+```
+
+</content>
+</details>
+
+<hr/>
+
+## Default Templates
```
Comment (ivan-burazin) [inline src/content/docs/configuration/templates.mdx:176]: Here should be explained that there is a default template index. where it is and what its made of. And that the user should copy that and edit to create their own. And walk them through the process

Comment (ivan-burazin) [reply src/content/docs/configuration/templates.mdx:176]: After you explain the index. then you should explain how the repo of the each template looks. lastly now that the user has a new index and new templates how to deploy that back to their daytona

**Review by ivan-burazin — CHANGES_REQUESTED** [review]
Quite a few changes needed. Watch https://www.youtube.com/watch?v=7UUFDkfFmu4 for more details.

File: src/content/docs/configuration/templates.mdx (hunk 3 above)
Comment (zmzlois) [reply src/content/docs/configuration/templates.mdx:107]: should this move to the /usage/workspace ? it was very headachy to do after creating workspace from template

File: src/content/docs/configuration/templates.mdx (hunk 2 above)
Comment (zmzlois) [reply src/content/docs/configuration/templates.mdx:34]: noted

File: src/content/docs/configuration/templates.mdx (hunk 1 above)
Comment (zmzlois) [reply src/content/docs/configuration/templates.mdx:17]: I will move that paragraph above and remove this

**Pull request closed without merging**